plugins/api/src/atomic_apply_acl_operations.cpp

```diff
@@ -207,14 +216,12 @@ namespace
 
     auto to_access_type_id(std::string_view _acl) -> int
     {
-        // clang-format off
-        if (_acl == "read")  { return 1050; }
-        if (_acl == "write") { return 1120; }
-        if (_acl == "own")   { return 1200; }
-        if (_acl == "null")  { return 1000; }
-        // clang-format on
-
-        THROW(SYS_INVALID_INPUT_PARAM, fmt::format("Invalid ACL [acl={}]", _acl));
+        if (acl_to_id.find(_acl) == acl_to_id.cend()) {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+            THROW(SYS_INVALID_INPUT_PARAM, fmt::format("Invalid ACL [acl={}]", _acl));
+        }
+
+        return acl_to_id.at(_acl);
```

Comment:
Minor improvement: We could store the iterator returned by `find` and use that instead of calling `at` in the `return` line to save an extra search.

Otherwise, we could use `throw_if_invalid_acl` here since L219-L222 pretty much duplicates that function.

Reply:
Oh, I just noticed your question. Please ignore suggestion as appropriate